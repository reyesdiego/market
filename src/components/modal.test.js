import React from "react";
import { shallow, configure } from "enzyme";
import Modal from "./modal";

import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe("Modal", () => {
  it("renders nothing when closed", () => {
    expect(shallow(<Modal onClose={jest.fn()} />)).toMatchSnapshot();
  });

  it("renders when open", () => {
    expect(
      shallow(<Modal show={true} onClose={jest.fn()} />)
    ).toMatchSnapshot();
  });

  it("calls onClose when button is clicked and closeable", () => {
    const onClose = jest.fn();
    const wrapper = shallow(
      <Modal show={true} onClose={onClose} closeable={true} />
    );

    const btn = wrapper.find("button");
    btn.simulate("click");
    expect(onClose).toBeCalled();
  });
});
