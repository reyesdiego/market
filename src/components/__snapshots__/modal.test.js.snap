// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders nothing when closed 1`] = `""`;

exports[`Modal renders when open 1`] = `
<div
  style={
    Object {
      "backgroundColor": "rgba(0,0,0,0.3)",
      "bottom": 0,
      "left": 0,
      "padding": 50,
      "position": "fixed",
      "right": 0,
      "top": 0,
    }
  }
>
  <div
    style={
      Object {
        "backgroundColor": "#fff",
        "borderRadius": 5,
        "margin": "0 auto",
        "maxWidth": 300,
        "padding": 30,
      }
    }
  />
</div>
`;
